### ISSUE TITLE
Allow `Column` type for timezone argument in pyspark.sql.functions 

### ISSUE THREAD

**Issue by charlietsai** [zero323/pyspark-stubs#380]
In the functions here:
https://github.com/zero323/pyspark-stubs/blob/3c4684a224c1be4eea4577e475f8bb4d045edddd/third_party/3/pyspark/sql/functions.pyi#L100-L101
we currently have `tz: str` but this can also be specified as a `Column`

Example:
```python
>>> from pyspark.sql import functions
>>> df = spark.sql("SELECT CAST(0 AS TIMESTAMP) AS timestamp, 'Asia/Tokyo' AS tz")
>>> df.select(functions.from_utc_timestamp(df.timestamp, df.tz)).collect()
[Row(from_utc_timestamp(timestamp, tz)=datetime.datetime(1970, 1, 1, 18, 0))]
```

I think this could be expanded to `tz: ColumnOrName`?

**Comment by zero323** [comment]
Thanks for reporting @charlietsai. Care to open a PR for this?

> I think this could be expanded to `tz: ColumnOrName`?

Indeed, although in this particular case I'd prefer to have explicit `Union[Column, str]` to stress out that meaning is different from `ColumnOrName`. It won't make any difference for the type checker, but can make a big one for anyone who reads the annotation.

**Comment by charlietsai** [comment]
Makes sense to me! Opening a PR

**Comment by zero323** [comment]
Thank you @charlietsai. Merged into master and backported to branch-3.0 and branch-2.4

**Comment by charlietsai** [comment]
Thanks @zero323 for this work!
